### PR TITLE
add simple command list filter

### PR DESCRIPTION
**Pull request by luthermonson**
Was getting annoyed by typoing govc commands and only getting a giant list returned and having to scroll and find what I wanted.

example usage
```➜  govc git:(cmd-list-filter) ./govc-dev tags
Usage of ./govc-dev filtered by tags
  disk.tags.attach
  disk.tags.detach
  tags.attach
  tags.attached.ls
  tags.category.create
  tags.category.info
  tags.category.ls
  tags.category.rm
  tags.category.update
  tags.create
  tags.detach
  tags.info
  tags.ls
  tags.rm
  tags.update
```

### CODE REVIEW

File: govc/cli/command.go [hunk 1]
```diff
@@ -43,16 +44,25 @@ type Command interface {
 	Run(ctx context.Context, f *flag.FlagSet) error
 }
 
-func generalHelp(w io.Writer) {
-	fmt.Fprintf(w, "Usage of %s:\n", os.Args[0])
+func generalHelp(w io.Writer, filter string) {
+	if filter == "" {
+		fmt.Fprintf(w, "Usage of %s:\n", os.Args[0])
+	} else {
+		fmt.Fprintf(w, "Usage of %s filtered by %s\n", os.Args[0], filter)
```
Comment (dougm) [inline govc/cli/command.go]: :+1: this is useful.  Can we change the wording a bit though, maybe borrowing from ubuntu(?):

```% nope
No command 'nope' found, did you mean:
... list of matching commands ...
```

``` govc tags
govc: command 'tags' not found, did you mean:
  disk.tags.attach
  disk.tags.detach
  tags.attach
  tags.attached.ls
  tags.category.create
  tags.category.info
  tags.category.ls
  tags.category.rm
  tags.category.update
  tags.create
  tags.detach
  tags.info
  tags.ls
  tags.rm
  tags.update
```

Comment (luthermonson) [reply govc/cli/command.go]: yup easy peasy

**Comment by luthermonson** [comment]
@dougm updated and force pushed.

**Comment by dougm** [comment]
@luthermonson just noticed that no commands are listed if nothing matches.  For example, I just did:
```console
% govc -vim-version test
No command '-vim-version' found, did you mean:
```

Ideally it would list the full list of commands if nothing matches as it did previously.

**Pull request merged**